app/Http/Controllers/CourseCompletedController.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use App\Models\CourseCompleted;
+use Carbon\Carbon;
+
+class CourseCompletedController extends Controller
+{
+    /**
+     * Display a listing of the course completions.
+     */
+    public function index(Request $request)
+    {
+           // First, sync data from attendance to course_completed
+           $this->syncCompletionsFromAttendance();
+
+        // Fetch all course completions with user and course information
+        $courseCompletions = DB::table('course_completed')
+            ->join('users', 'course_completed.user_id', '=', 'users.userId')
+            ->join('courses', 'course_completed.course_id', '=', 'courses.id')
+            ->select('course_completed.*', 'users.name as user_name', 'courses.course_name as course_title')
+            ->orderBy('course_completed.completed_at', 'desc')
+            ->get();
+
+
+
+        return view('admin.course_completed', compact('courseCompletions'));
+    }
+
+    /**
+     * Delete a course completion record.
+     */
+    private function syncCompletionsFromAttendance()
+    {
+        // Find users who have attended courses more than 10 times
+        $completions = DB::table('attendances')
```

Comment:
Rather than fetch the data here, create a function getAttendanceForCourse($course_id) that will return the total number of attendance for a user for a particular course, it will be reused